src/train/extractor/extractor.py

```diff
@@ -177,6 +201,31 @@ def get_system_feature_data(self, query_results, features):
             feature_data_list += [aggr_query_data]
         feature_data = pd.concat(feature_data_list, axis=1).astype(int)
         return feature_data
+    
+    # get accelerator feature data and process (current only intel_qat)
+    def get_accelerator_feature_data(self, query_results, features):
+        feature_data_list = []
+        cur_accelerator_features = []
+        for feature in features:
+            query = feature_to_query(feature)
+            if query not in query_results:
+                print(query, "not in", list(query_results.keys()))
+                return None, None
+            aggr_query_data = query_results[query].copy()
+            tmp_list=[]
+            accelerator_type_df = aggr_query_data.groupby([accelerator_value_type_colname])
+            for type_name, type_data in accelerator_type_df:
+                #convert colname to a combination of feature and numerical types
+                new_colname = "{}_{}".format(feature, type_name)
+                type_data.rename(columns={query: new_colname}, inplace=True)
+                type_data = type_data [[TIMESTAMP_COL, new_colname]]
+                type_data = type_data .groupby([TIMESTAMP_COL]).sum().sort_index()
+                tmp_list += [type_data ]
+                cur_accelerator_features.append(new_colname)
+            tmp_data_list = pd.concat(tmp_list,axis=1).fillna(0).astype(int).sort_values(by=TIMESTAMP_COL)
+            feature_data_list += [tmp_data_list]
+        feature_data = pd.concat(feature_data_list, axis=1).astype(int)
+        return feature_data, cur_accelerator_features
```

Comment:
Why we need to separate `accelerator_features` from other feature_data?

Reply:
It's node level. And in kepler now, the metrics is : 
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev0",source="intel-qat",type="latency"} 46750
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev1",source="intel-qat",type="latency"} 0
May will add more like:
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev0",source="intel-qat",type="compress"} 4
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev1",source="intel-qat",type="compress"} 1
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev0",source="intel-qat",type="decompress"} 4
kepler_node_accelerator_intel_qat{instance="cpio-5",qatDevID="qat_dev1",source="intel-qat",type="decompress"} 1
Just different types, so I also need to distinguish between different types

Reply:
I see. So, it is the metric that is available only on the node level. 

I think instead of separating it like this. We should modify the extractor to add a separated function to extract the node-level metrics which is not limited to only this `accelerator_features`. Also, we should support extracting multiple metrics from a single query like this case in the future. This could be a refactoring the later PR. 

For this PR, 
1. we need to add condition `node_level==True` to check whether the accelerator_features could be applied.
2. You can set `fg = FeatureGroup[feature_group]`, then check `fg == FeatureGroup.AcceleratorOnly` directly. 
3. Should define feature name per column. Not separate it to `cur_accelerator_features`.  (please remove additional return from extract function)

      ```python
      ACCELERATE_FEATURES = ["accelerator_intel_qat_latency", "accelerator_intel_qat_compress", "accelerator_intel_qat"_decompress"]
      ```
      
4. rebase from the main branch